File: Problemset/palindrome-number/palindrome-number.py

# @Title: 回文数 (Palindrome Number)
# @Author: 18015528893
# @Date: 2019-10-21 21:41:32
# @Runtime: 84 ms
# @Memory: 13.7 MB

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x == 0:
            return True
        if x < 0:
            return False
        if x % 10 == 0:
            return False
        a = list()
        while True:
            a.append(x % 10)
            x = x // 10
            if x == 0:
                break
        if a[::-1] == a:
            return True
        return False

File: Problemset/palindrome-number/palindrome-number.java

// @Title: 回文数 (Palindrome Number)
// @Author: 18015528893
// @Date: 2021-02-18 18:59:21
// @Runtime: 10 ms
// @Memory: 37.9 MB

class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        int ans = 0;
        int i = x;
        while (i != 0) {
            if (i > 0 && ans > (Integer.MAX_VALUE - i % 10) / 10) {
                return false;
            }
            if (i < 0 && ans < (Integer.MIN_VALUE - i % 10) / 10) {
                return false;
            }
            ans = 10 * ans + i % 10;
            i /= 10;
        }
        if (x == ans) {
            return true;
        } else {
            return false;
        }
    }
}
